appsmithorg/appsmith
# feat: Fetch Appsmith version when browser tab comes on focus

## Description
Problem:
- When Appsmith upgrades, the older version JS files are no longer hosted on the server.
- When a browser tab with Appsmith is left open for a long time and not in focus, and Appsmith has upgraded in the mean time, the client will request for the older JS files. In this case, an html file is returned by Nginx while the browser expected JS

Solution:
- Ping the RTS server about the tab being back in focus, which will then respond with the current version of Appsmith.
- Version match will kick in on the client and if they don't match, it will ask the user to refresh the page

Fixes #9740 
Fixes #16235


## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manual Test plan:
Run RTS and Appsmith client locally
Switch to another tab on the browser
Update the version number on the RTS code (version.js)
Come back to the Appsmith tab

Assert: A toast is shown to indicate a refresh is needed to update the client


## Checklist:
